lib/suite.js

```diff
@@ -420,6 +420,79 @@ Suite.prototype.eachTest = function(fn) {
  */
 Suite.prototype.run = function run() {
   if (this.root) {
-    this.emit('run');
+    this.emit(constants.SUITE_EVENT_RUN);
   }
 };
+
+/**
+ * Returns the array of hooks by hook name; see `SUITE_HOOK_*` constants.
+ * @private
+ */
+Suite.prototype.getHooks = function getHooks(name) {
+  return this['_' + name];
+};
+
+/**
+ * Cleans up the references to all the deferred functions
+ * (before/after/beforeEach/afterEach) and tests of a Suite.
+ * These must be deleted otherwise a memory leak can happen,
+ * as those functions may reference variables from closures,
+ * thus those variables can never be garbage collected as long
+ * as the deferred functions exist.
+ *
+ * @private
+ */
+Suite.prototype.cleanReferences = function cleanReferences() {
+  function cleanArrReferences(arr) {
+    for (var i = 0; i < arr.length; i++) {
+      delete arr[i].fn;
+    }
+  }
+
+  if (Array.isArray(this._beforeAll)) {
+    cleanArrReferences(this._beforeAll);
+  }
+
+  if (Array.isArray(this._beforeEach)) {
+    cleanArrReferences(this._beforeEach);
+  }
+
+  if (Array.isArray(this._afterAll)) {
+    cleanArrReferences(this._afterAll);
+  }
+
+  if (Array.isArray(this._afterEach)) {
+    cleanArrReferences(this._afterEach);
+  }
+
+  for (var i = 0; i < this.tests.length; i++) {
+    delete this.tests[i].fn;
+  }
```

Comment:
Realize you just migrated the existing code. Question unrelated to this PR.
* Should we be using `delete` here instead of `this.tests[i].fn = null;`?

While `delete` is better semantically, (re)setting the value is [far more performant](https://jsperf.com/delete-vs-undefined-vs-null/16) assuming no aliasing. As this is being done _only_ for GC purposes, I'm glossing over the _actual_ difference between the two and only looking at the performance.

See also:
* [Does using delete keyword effect v8 optimizations of an object?]
(https://groups.google.com/forum/#!topic/v8-users/zE4cOHBkAnY)
* [Trash talk: the Orinoco GC](https://v8.dev/blog/trash-talk)
* [How To Write Fast, Memory-Efficient JavaScript](https://www.smashingmagazine.com/2012/11/writing-fast-memory-efficient-javascript/)

Reply:
I don't know.. I didn't write this stuff.  if you want to profile it and have a better solution, please send a PR